test/integration/scheduler/nominated_node_name/nominated_node_name_test.go

```diff
@@ -0,0 +1,843 @@
+			if err != nil {
+				t.Fatalf("delete victim pod: %v", err)
+			}
+
+			err = testutils.WaitForPodToSchedule(testCtx.Ctx, testCtx.ClientSet, preemptor)
+			if err != nil {
+				t.Fatalf("Preemptor pod failed to schedule: %v", err)
+			}
+
+			preemptor, err = testCtx.ClientSet.CoreV1().Pods(preemptor.Namespace).Get(testCtx.Ctx, preemptor.Name, metav1.GetOptions{})
+			if err != nil {
+				t.Fatalf("get scheduled pod: %v", err)
+			}
+
+			if enabled {
+				if preemptor.Status.NominatedNodeName != "" {
+					t.Errorf("NominatedNodeName not cleared: %s", preemptor.Status.NominatedNodeName)
+				}
+			} else {
+				if preemptor.Status.NominatedNodeName == "" {
+					t.Errorf("NominatedNodeName was cleared")
+				}
+			}
+
+		})
+	}
+}
+
+// TestExternalComponentSetsNominatedNodeName tests scheduler handling of externally set NominatedNodeName.
+func TestExternalComponentSetsNominatedNodeName(t *testing.T) {
+	testCtx := testutils.InitTestSchedulerWithNS(t, "external-nnn-test")
+
+	nodeNames := createNodesFromSpecs(t, testCtx.ClientSet, []testNode{
+		{name: "node1"},
+		{name: "node2"},
+	})
+
+	targetNodeName := nodeNames[1]
+	pod := st.MakePod().Name("external-nnn-pod").Namespace(testCtx.NS.Name).
+		Container("c").
+		SchedulingGates([]string{"example.com/external-nnn-test"}).
+		Obj()
+
+	createdPod, err := testCtx.ClientSet.CoreV1().Pods(testCtx.NS.Name).Create(testCtx.Ctx, pod, metav1.CreateOptions{})
+	if err != nil {
+		t.Fatalf("create pod: %v", err)
+	}
+
+	createdPod.Status.NominatedNodeName = targetNodeName
+	_, err = testCtx.ClientSet.CoreV1().Pods(testCtx.NS.Name).UpdateStatus(testCtx.Ctx, createdPod, metav1.UpdateOptions{})
+	if err != nil {
+		t.Fatalf("update pod status: %v", err)
+	}
+
+	updatedPod, err := testCtx.ClientSet.CoreV1().Pods(testCtx.NS.Name).Get(testCtx.Ctx, createdPod.Name, metav1.GetOptions{})
+	if err != nil {
+		t.Fatalf("get pod: %v", err)
+	}
+	if updatedPod.Status.NominatedNodeName != targetNodeName {
+		t.Fatalf("NominatedNodeName mismatch: want %s, got %s", targetNodeName, updatedPod.Status.NominatedNodeName)
+	}
+
+	updatedPod.Spec.SchedulingGates = nil
+	_, err = testCtx.ClientSet.CoreV1().Pods(testCtx.NS.Name).Update(testCtx.Ctx, updatedPod, metav1.UpdateOptions{})
+	if err != nil {
+		t.Fatalf("remove scheduling gate: %v", err)
+	}
+
+	err = testutils.WaitForPodToSchedule(testCtx.Ctx, testCtx.ClientSet, updatedPod)
+	if err != nil {
+		t.Fatalf("schedule pod %s: %v", updatedPod.Name, err)
+	}
+	scheduledPod, err := testCtx.ClientSet.CoreV1().Pods(testCtx.NS.Name).Get(testCtx.Ctx, updatedPod.Name, metav1.GetOptions{})
+	if err != nil {
+		t.Fatalf("get scheduled pod %s: %v", updatedPod.Name, err)
+	}
+
+	if scheduledPod.Spec.NodeName != targetNodeName {
+		t.Errorf("want node %s, got %s", targetNodeName, scheduledPod.Spec.NodeName)
+	}
+}
+
+// TestHighPriorityIgnoresNominatedNode tests high priority pod scheduling to available nodes.
+func TestHighPriorityIgnoresNominatedNode(t *testing.T) {
```

Comment:
This is just a simple preemption scenario right now, right?

I guess you meant to create this test case:
> The scheduler ignores NominatedNodeName reservations on pods when it's scheduling higher priority pods.
https://github.com/kubernetes/enhancements/tree/master/keps/sig-scheduling/5278-nominated-node-name-for-expectation#integration-tests

If so, you have to put NNN on a low priority pod, schedule a high priority pod (w/o NNN), and see if the high priority pod can steal the place of a low priority pod that is reserved by NNN.